src/main/java/Duke.java

```diff
@@ -1,10 +1,137 @@
-        System.out.println("Hello from\n" + logo);
+        Task[] taskList = new Task[MAX_NO_OF_TASKS];
+        taskCount = 0;
+        String userInput;
+        Scanner in = new Scanner(System.in);
+
+        sayIntro();
+        //easier to identify lines input by user (per Python)
+        System.out.print(">>>");
+        
+        while (in.hasNextLine()) {
+            userInput = in.nextLine();
+            String[] tokenizedInput = userInput.split(" ");
+            if (tokenizedInput[0].equals("bye")) {
+                break;
+            } else if (tokenizedInput[0].equals("list")) {     
+                addTaskToList(taskList);
+            } else if (tokenizedInput[0].equals("done")) {
+                updateTaskDone(tokenizedInput[1], taskList);
+            } else {
+                insertNewTask(taskList, userInput, tokenizedInput);
+            }
+            System.out.print(">>>");
+        }
+
+        sayGoodbye();
+    }
+
+    public static void addTaskToList(Task[] listInput) {
+        //if list empty, inform user and await next command
+        if (taskCount == 0) {
+            System.out.println(underscoredLine + System.lineSeparator() + "\tThe list is empty." 
+            + System.lineSeparator() + underscoredLine);
+            return;
+        }
+        //if list non-empty, print out all existing tasks
+        System.out.println(underscoredLine + System.lineSeparator());
+        for (int i = 0; i < taskCount; i++) {
+            System.out.println("\t" + Integer.toString(i + 1) + "." + listInput[i].toString());
+        }
+        System.out.println(underscoredLine);
+        return;
+    }
+
+    public static void updateTaskDone(String taskNumberInput, Task[] listInput){
+        int queryNumber = Integer.parseInt(taskNumberInput);
+        boolean isOutOfRange = queryNumber < 1 || queryNumber > taskCount;
+        
+        //handle case where user inputs non-existing task number to mark as done
+        if (isOutOfRange){
+            System.out.println(underscoredLine + System.lineSeparator() + "\tInvalid task number."
+            + System.lineSeparator() + underscoredLine);
+            return;
+        }
+        //handle case where user tries to mark as done an already completed task
+        boolean isTaskAlreadyDone = listInput[queryNumber-1].getIsDone();
+        if (isTaskAlreadyDone){
+            System.out.println(underscoredLine + System.lineSeparator()
+            + "\tThis task has already been marked completed." + System.lineSeparator() + underscoredLine);
+            return;
+        }
+        listInput[queryNumber-1].markAsDone();
+        System.out.println(underscoredLine + System.lineSeparator() + "\tGreat job! I've marked this task as done:\n"
+        + "\t" + Integer.toString(queryNumber) + ".[" + listInput[queryNumber-1].getStatusIcon() + "] "
+        + listInput[queryNumber-1].getDescription() + System.lineSeparator() + underscoredLine);
```

Comment:
Consider shortening the line, into String s = ... Then do a few s += ... to split up the different component so that it is more readable and easier to understand. Finally then print the String s. Also, for wrapped text, consider indenting 8 spaces more than the parent line